pandas/core/dtypes/common.py

```diff
@@ -601,7 +601,8 @@ def is_dtype_equal(source, target) -> bool:
             #  is_dtype_equal(CDT, "category") and CDT == "category"
             try:
                 src = get_dtype(source)
-                if isinstance(src, ExtensionDtype):
+                if isinstance(src, ExtensionDtype) or isinstance(src, np.dtype):
```

Comment:
can you add the example to the doc-string

Reply:
Done

Reply:
single isinstance call `isinstance(src, (ExtensionDtype, np.dtype))`

Reply:
Done